packages/react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -149,11 +150,14 @@ class Checkbox extends AutoControlledComponent<WithAsProp<CheckboxProps>, Checkb
     })
 
     return (
-      <ElementType
-        className={classes.root}
+      <CheckboxBase
+        slots={{ root: ElementType }}
+        slotProps={{ input: { style: { visibility: 'hidden', display: 'none' } } }}
```

Comment:
You might want to pull this out of the render function in a `const` object so that this isn't recomputed on every render.